packages/fxa-payments-server/src/components/CouponForm/index.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, MouseEventHandler, FormEventHandler } from 'react';
+import { Localized } from '@fluent/react';
+import './index.scss';
+import { Coupon } from '../../lib/Coupon';
+
+type CouponFormProps = {
+  coupon?: Coupon;
+  setCoupon: (coupon: Coupon | undefined) => void;
+};
+
+export const CouponForm = ({ coupon, setCoupon }: CouponFormProps) => {
+  const [hasCoupon, setHasCoupon] = useState(coupon ? true : false);
+  const [couponCode, setCouponCode] = useState(coupon ? 'test' : '');
+  const [error, setError] = useState(false);
+
+  const onSubmit: FormEventHandler = (event: any) => {
```

Comment:
We shouldn't assert `event` to be `any`; we could just rely on `FormEventHandler`'s type, so you could just remove the type declaration for `event` and it'd have the correct type.